tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -339,23 +339,15 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
                                         .WithKeys(ConfigurationKeys.AzureAppService.AzureAppServicesContextKey)
                                         .AsBool(false);
 
-            // we only want to instantiate ImmutableAzureAppServiceSettings if we are for sure running
-            // in AAS, or if we might be running in a Consumption Plan function.
-            var isMaybeAzureFunction = config.WithKeys(ConfigurationKeys.AzureAppService.FunctionsExtensionVersionKey).AsString() is not null;
-            // consumption plan Azure functions will have WEBSITE_SKU=Dynamic, or will be missing the var altogether.
-            // app service plan functions will always have a WEBSITE_SKU value, and it's never Dynamic.
-            var websiteSKUDynamicOrMissing = config.WithKeys(ConfigurationKeys.AzureAppService.WebsiteSKU).AsString("Dynamic") is "Dynamic" or null;
-            var isMaybeAzureFunctionConsumptionPlan = isMaybeAzureFunction && websiteSKUDynamicOrMissing;
+            IsRunningInAzureFunctionsConsumptionPlan = ImmutableAzureAppServiceSettings.GetIsFunctionsAppConsumptionPlan(source, telemetry);
 
-            if (IsRunningInAzureAppService || isMaybeAzureFunctionConsumptionPlan)
+            if (IsRunningInAzureAppService)
```

Comment:
Does this still need to be `if (IsRunningInAzureAppService || IsRunningInAzureFunctionsConsumptionPlan)` ? 🤔

Reply:
not anymore, since we can get `IsRunningInAzureFunctionsConsumptionPlan` from the helper now